app/controllers/application_controller/ci_processing.rb

```diff
@@ -828,7 +828,7 @@ def discover
       drop_breadcrumb(:name => _("%{title} Discovery") % {:title => title}, :url => "/host/discover")
       @discover_type_selected = params[:discover_type_selected]
 
-      if request.parameters[:controller] == "ems_cloud" && params[:discover_type_selected] == ExtManagementSystem.ems_cloud_discovery_types['azure']
+      if request.parameters[:controller] == "ems_cloud" && params[:discover_type_selected] == 'azure'
```

Comment:
So, I'd think that round 2 of all of this refactoring would be to remove the `== 'azure'` altogether and have some generic way to ask about the fields referenced below.